tensorflow_addons/losses/triplet.py

```diff
@@ -97,7 +100,14 @@ def triplet_semihard_loss(
     labels = tf.reshape(labels, [lshape[0], 1])
 
     # Build pairwise squared distance matrix.
-    pdist_matrix = metric_learning.pairwise_distance(precise_embeddings, squared=True)
+
+    if angular:
```

Comment:
Maybe other people can join in on this, should we add support for custom distance method? (probably easy to implement, 1 more arg and another `if` statement here)

Reply:
I agree that would be helpful, but how do you suggest the custom distance method is passed to the function. Do you mean that the function itself would be passed as an argument?

Reply:
We can do this similar to TensorFlow API that this parameter can take in either a string (eud/angular like you mentioned) and a function. For example when you build the model you can either have loss = some string or some class, so I agree with you here.